### PR TITLE
forced whitelist: use configuable kick message (fixes #5417)

### LINKED ISSUE #5417: Displaying wrong whitelist message

### PR DESCRIPTION
**Pull request by Trigary**
Fixes #5417

### CODE REVIEW

File: Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index c372e6897c73fad6f9c478cec9f58303fb0cce17..77f8be879ba098aaa15a5fd947dc3a4b39c5e397 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -2030,7 +2030,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+                 EntityPlayer entityplayer = (EntityPlayer) iterator.next();
+ 
+                 if (!whitelist.isWhitelisted(entityplayer.getProfile())) {
+-                    entityplayer.playerConnection.disconnect(new ChatMessage("multiplayer.disconnect.not_whitelisted"));
```
Comment (yannicklamprecht) [inline Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch:15]: Sounds like someone at Mojang forgot to create a translatable component.
Without the need of a custom message I would have preferred replacing it with a translatable component.
Due the need of a non plugin custom message, it's fine how it is handled in this pr.

**Review by yannicklamprecht — APPROVED** [review]

File: Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch (hunk 1 above)
Comment (zml2008) [reply Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch:15]:
```json
  "multiplayer.disconnect.not_whitelisted": "You are not white-listed on this server!",
```

seems to exist in en_us.json

Comment (electronicboy) [reply Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch:15]: That is a translatable component, just how mojang works is weird, according to the issue that all works as intended, just, they wish the whitelist message feature carried over

File: Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
++                    entityplayer.playerConnection.disconnect(org.spigotmc.SpigotConfig.whitelistMessage);
```
Comment (electronicboy) [inline Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch]: // Paper

**Review by electronicboy — APPROVED** [review]

File: Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch (hunk 2 above)
Comment (Trigary) [reply Spigot-Server-Patches/0698-forced-whitelist-use-configurable-kick-message.patch]: Ty, don't know how I managed to forget that.

**Pull request merged**